common/persistence/client/health_request_rate_limiter.go

```diff
@@ -149,19 +157,36 @@ func (rl *HealthRequestRateLimiterImpl) maybeRefresh() {
 }
 
 func (rl *HealthRequestRateLimiterImpl) refreshRate() {
+	rl.curOptions.RLock()
+	curOptions := rl.curOptions.options
+	rl.curOptions.RUnlock()
```

Comment:
Why not just use an atomic.Pointer to the curOptions structure? Then we could just load/store the struct and not need to care about locking semantics as this doesn't look to be modified, only replaced

Reply:
Yeah that is what I would do. It would also eliminate the need for the separate atomic boolean for enabled.

Reply:
Hm, that could work too. I think the main benefit of having this struct instead of `atomic.Pointer` is the `digest` that I introduced to reduce getting a write lock when there's no change to the dynamic config. But `atomic.Pointer` maybe is fast enough that won't matter.